landlab/grid/hex.py

```diff
@@ -118,6 +119,14 @@ def __init__(
         >>> hmg.number_of_nodes
         7
         """
+
+        # Pre-LL2.0: handle shape and node_layout keyword args
+        # (note: in LL2.0, shape should refer ONLY to rows and columns)
+        if type(shape) is str:  # "old" (LL < 2.0) usage
+            node_layout = shape
+        elif type(shape) is tuple:
+            (base_num_rows, base_num_cols) = shape
+
```

Comment:
@gregtucker I think if we want this to be in current LL (as opposed to LL2.0) we need to not have lines 127 and 128... Doesn't that permit someone to pass shape as tuple and then have it used as the nr and nc?

Reply:
Yes, but I don't think that's a problem. It supports both old usage, when shape is a str, and new usage, when shape is a tuple. In other words, it should still be fully compatible with old usage, i.e., is still backward compatible.